public/app/plugins/panel/gauge/GaugePanel.tsx

```diff
@@ -54,6 +54,19 @@ export class GaugePanel extends PureComponent<PanelProps<PanelOptions>> {
 
   getValues = (): FieldDisplay[] => {
     const { data, options, replaceVariables, fieldConfig, timeZone } = this.props;
+
+    for (let frame of data.series) {
+      for (let field of frame.fields) {
+        // Set the Min/Max value automatically for percent and percentunit
+        if (field.config.unit === 'percent' || field.config.unit === 'percentunit') {
+          const min = field.config.min ?? 0;
+          const max = field.config.max ?? field.config.unit === 'percent' ? 100 : 1;
```

Comment:
I think brackets after the `??` operator are needed, otherwise it checks the entire left side of ternary for truthiness and always returns either 100 or 1, but never `config.max`, even when it's set. e.g:

`const max = field.config.max ?? (field.config.unit === 'percent' ? 100 : 1);`